### PR TITLE
build: upgrade image-spec to v1.1.0-rc.3

### LINKED ISSUE #495: MediaTypeArtifactManifest not found in v2
Hi there, 
After updating to V2, linter failed with 

```bash
    golangci-lint run ./...        
```

```
WARN [runner] Can't run linter goanalysis_metalinter: buildir: failed to load package descriptor: could not load export data: no export data for "oras.land/oras-go/v2/internal/descriptor" 
ERRO Running error: 1 error occurred:
        * can't run linter goanalysis_metalinter: buildir: failed to load package descriptor: could not load export data: no export data for "oras.land/oras-go/v2/internal/descriptor"
```

after dig into the code, I noticed this const ( `MediaTypeArtifactManifest`) has not been set in the code.

### PR DESCRIPTION
**Pull request by shizhMSFT**
This PR fixes and only fixes the compiling errors due to the breaking changes introduced by `image-spec v1.1.0-rc.3` where the artifact manifest related specs are removed.

Fix #495.
Fix partially #494

### CODE REVIEW

File: internal/spec/artifact.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+// AnnotationReferrersFiltersApplied is the annotation key for the comma separated list of filters applied by the registry in the referrers listing.
+const AnnotationReferrersFiltersApplied = "org.opencontainers.referrers.filtersApplied"
+
+// MediaTypeArtifactManifest specifies the media type for a content descriptor.
+const MediaTypeArtifactManifest = "application/vnd.oci.artifact.manifest.v1+json"
+
+// Artifact describes an artifact manifest.
+// This structure provides `application/vnd.oci.artifact.manifest.v1+json` mediatype when marshalled to JSON.
+type Artifact struct {
```
Comment (rogpeppe) [inline internal/spec/artifact.go:28]: Drive-by comment: I wonder if it might be good to document the status of this media type here (i.e. that it's not part of the current spec and only here for Go compatibility, maybe with a link to somewhere with more info) ?

Comment (shizhMSFT) [reply internal/spec/artifact.go:34]: Documented the status of the manifest type.

**Review by yuseferi — APPROVED** [review]

**Review by TerryHowe — APPROVED** [review]
LGTM

**Review by SteveLasker — APPROVED** [review]
LGTM
